sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -270,12 +271,25 @@ trait CheckAnalysis extends PredicateHelper {
 
         operator match {
           case o if o.children.nonEmpty && o.missingInput.nonEmpty =>
+            val resolver = plan.conf.resolver
+            val attrsWithSameName = o.missingInput.filter(x =>
+              o.inputSet.exists(y => resolver(x.name, y.name)))
             val missingAttributes = o.missingInput.mkString(",")
-            val input = o.inputSet.mkString(",")
+            val availableAttributes = o.inputSet.mkString(",")
+            val repeatedNameHint = if (attrsWithSameName.size > 0) {
+              val commonNames = attrsWithSameName.map(_.name).mkString(",")
+              s"""\n|Attribute(s) `$commonNames` seem to appear in two
+                  |different datasets, with the same name."""
```

Comment:
The `datasets` concept is a little strange here, would `inputs` or `input operators` be better?

Reply:
`\n|Please check attribute(s) `$commonNames`, they seem to appear in two...`

Reply:
I have changed it. Agree datasets may not be the best word to define it